builder/common/step_network_info_test.go

```diff
@@ -0,0 +1,409 @@
+				}
+
+				vpcFilterFound = true
+			}
+
+			if !vpcFilterFound {
+				return nil, fmt.Errorf("expected to find vpc-id filter, but did not find it")
+			}
+
+			return &ec2.DescribeSubnetsOutput{
+				Subnets: []*ec2.Subnet{
+					{
+						AvailabilityZone:        aws.String("us-east-1a"),
+						SubnetId:                aws.String("subnet1"),
+						AvailableIpAddressCount: aws.Int64(256),
+					},
+					{
+						AvailabilityZone:        aws.String("us-east-1b"),
+						SubnetId:                aws.String("subnet2"),
+						AvailableIpAddressCount: aws.Int64(512),
+					},
+				},
+			}, nil
+		},
+		describeInstanceTypeOfferings: func(in *ec2.DescribeInstanceTypeOfferingsInput) (*ec2.DescribeInstanceTypeOfferingsOutput, error) {
+			if *in.LocationType != "availability-zone" {
+				return nil, fmt.Errorf("called DescribeInstanceTypeOfferings with LocationType = %q, expected availability_zone", *in.LocationType)
+			}
+
+			var machines []*ec2.InstanceTypeOffering
+
+			foundLocation := false
+			for _, filter := range in.Filters {
+				if *filter.Name != "location" {
+					continue
+				}
+				foundLocation = true
+
+				filterVal := *filter.Values[0]
+				switch filterVal {
+				case "us-east-1a":
+					machines = []*ec2.InstanceTypeOffering{
+						{
+							InstanceType: aws.String("t2.mini"),
+						},
+						{
+							InstanceType: aws.String("t3.large"),
+						},
+					}
+				case "us-east-1b":
+					machines = []*ec2.InstanceTypeOffering{
+						{
+							InstanceType: aws.String("t2.mini"),
+						},
+						{
+							InstanceType: aws.String("t2.micro"),
+						},
+					}
+				default:
+					return nil, fmt.Errorf("error: location %q not expected", filterVal)
+				}
+			}
+
+			if !foundLocation {
+				return nil, fmt.Errorf("couldn't find location in filters")
+			}
+
+			return &ec2.DescribeInstanceTypeOfferingsOutput{
+				InstanceTypeOfferings: machines,
+			}, nil
+		},
+	}
+
+	stepConfig := &StepNetworkInfo{
+		AssociatePublicIpAddress: confighelper.TriTrue,
+		RequestedMachineType:     "t3.large",
+	}
+
+	state := &multistep.BasicStateBag{}
+	state.Put("ec2", mockConn)
+	state.Put("ui", &packersdk.MockUi{})
+
+	actRet := stepConfig.Run(context.Background(), state)
+	if actRet == multistep.ActionHalt {
+		t.Fatalf("running the step failed: %s", state.Get("error").(error))
+	}
+
+	vpcid, ok := state.GetOk("vpc_id")
+	if !ok || vpcid != "default-vpc" {
+		t.Errorf("error: vpc should be 'default-vpc', but is %q", vpcid)
+	}
+	t.Logf("set vpc is %q", vpcid)
+
+	subnetid, ok := state.GetOk("subnet_id")
+	if !ok || subnetid != "subnet1" {
+		t.Errorf("error: subnet should be 'subnet1', but is %q", subnetid)
+	}
+	t.Logf("set subnet is %q", subnetid)
+
+	az, ok := state.GetOk("availability_zone")
+	if !ok || az != "us-east-1a" {
+		t.Errorf("error: availability_zone should be 'us-east-1a', but is %q", az)
+	}
+	t.Logf("set AZ is %q", az)
+}
+
+func TestGetDefaultVPCFailDueToPermissions(t *testing.T) {
```

Comment:
Having these test cases are great.  Seeing as this is all mocked now I think we should add some test for testing when a VPC and Subnet Id is provided with and without an associate public ip to ensure the conditions don't get hit.

Reply:
Yes, there's a bunch of tests that we can write for the rest of the file as well, right now I focused on the changes I added with this PR.
Though you have a point in this case, we should make sure we don't trigger this condition. I'll set that up before we merge this PR.

Reply:
> Though you have a point in this case, we should make sure we don't trigger this condition. I'll set that up before we merge this PR.

Yeah seeing as this was the bug in v1.2.3 it seems only fair to test for it now.